src/PowerJoins.php

```diff
@@ -351,7 +351,18 @@ public function scopePowerJoinWhereHas(Builder $query, $relation, $callback = nu
      */
     public function relationshipAlreadyJoined($relation)
     {
-        return isset($this->joinRelationshipCache[spl_object_id($this)][$relation]);
+        $objectId = spl_object_id($this);
+        $beforeLast = Str::beforeLast(Str::replace('.0', '', $relation), '.');
```

Comment:
@luisdalmolin I wasn't sure why the first relationship adds the `".0"` in the `$relation` that is cached, but for comparison sake, I think it's safe to remove it to compare to other variants of the same cached `$relation`. Definitely hoping you can take a close look at this to see if this makes sense.